cmd/cstorvolumeclaim/controller.go

```diff
@@ -385,3 +391,180 @@ func BaseLabels(cvc *apis.CStorVolumeClaim) map[string]string {
 	}
 	return base
 }
+
+// cvcNeedResize returns true is a cvc requests a resize operation.
+func (c *CVCController) cvcNeedResize(cvc *apis.CStorVolumeClaim, cv *apis.CStorVolume) bool {
+
+	actualSize, err := resource.ParseQuantity(cv.Spec.Capacity)
+	if err != nil {
+		runtime.HandleError(err)
+		return false
+	}
+	requestSize := cvc.Spec.Capacity[v1.ResourceStorage]
+	return requestSize.Cmp(actualSize) > 0
+}
+
+// resizeCVC will:
+// 1. Mark cvc as resizing.
+// 2. Resize the cstorvolume object.
+// 3. Mark cvc as resizing finished
+func (c *CVCController) resizeCVC(cvc *apis.CStorVolumeClaim, cv *apis.CStorVolume) error {
+	if updatedCVC, err := c.markCVCResizeInProgress(cvc); err != nil {
+		glog.Errorf("Mark cvc %q as resizing failed: %v", cvc.Name, err)
+		return err
+	} else if updatedCVC != nil {
+		cvc = updatedCVC
+	}
+
+	// Record an event to indicate that cvc-controller is resizing this volume.
+	c.recorder.Event(cvc, corev1.EventTypeNormal, string(apis.CStorVolumeClaimResizing),
+		fmt.Sprintf("CVCController is resizing volume %s", cvc.Name))
+
+	err := func() error {
+		err := c.resizeVolume(cvc, cv)
+		if err != nil {
+			return err
+		}
+		// Resize volume succeeded mark it as resizing finished.
+		return c.markCVCResizeFinished(cvc)
```

Comment:
We need to wait for actual resize to complete.
Over here we should compare the actual and desired size in CV before updating this status.
A size field also needs to be added in CVC.